### PR TITLE
[formrecognizer] update sample readme

### PR DESCRIPTION
**Pull request by kristapratico**
Part of https://github.com/Azure/azure-sdk-for-python/issues/20841 https://github.com/Azure/azure-sdk-for-python/issues/20856

### CODE REVIEW

File: sdk/formrecognizer/azure-ai-formrecognizer/samples/README.md [hunk 1]
```diff
@@ -11,13 +11,37 @@ urlFragment: formrecognizer-samples
...
 
 All of these samples need the endpoint to your Form Recognizer resource ([instructions on how to get endpoint][get-endpoint-instructions]), and your Form Recognizer API key ([instructions on how to get key][get-key-instructions]).
 
+## Samples for client library versions 3.2.0b1 and later
+
+|**File Name**|**Description**|
+|----------------|-------------|
+|[sample_authentication.py][sample_auth] and [sample_authentication_async.py][sample_auth_async]|Authenticate the client|
+|[sample_analyze_layout.py][sample_analyze_layout] and [sample_analyze_layout_async.py][sample_analyze_layout_async]|Extract text, selection marks, and table structures in a document|
+|[sample_analyze_prebuilt_document.py][sample_analyze_prebuilt_document] and [sample_analyze_prebuilt_document_async.py][sample_analyze_prebuilt_document_async]|Analyze document key-value pairs, entities, tables, and selection marks using a prebuilt model|
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/samples/README.md:31]: name TBD

File: sdk/formrecognizer/azure-ai-formrecognizer/samples/README.md [hunk 2]
```diff
@@ -11,13 +11,37 @@ urlFragment: formrecognizer-samples
...
+|[sample_analyze_receipts.py][sample_analyze_receipts] and [sample_analyze_receipts_async.py][sample_analyze_receipts_async]|Analyze document text and pre-trained fields and values pertaining to English sales receipts using a prebuilt model|
+|[sample_analyze_custom_documents.py][sample_analyze_custom_documents] and [sample_analyze_custom_documents_async.py][sample_analyze_custom_documents_async]|Analyze custom documents with your custom model to extract text, field values, selection marks, and table data from documents|
+|[sample_build_model.py][sample_build_model] and [sample_build_model_async.py][sample_build_model_async]|Build a custom model|
+|[sample_create_composed_model.py][sample_composed_model] and [sample_create_composed_model_async.py][sample_composed_model_async]|Create a composed model from a collection of existing models to be called with a single model ID|
+|[sample_manage_models.py][sample_manage_models] and [sample_manage_models_async.py][sample_manage_models_async]|Manage the custom models in your account|
```
Comment (maririos) [inline sdk/formrecognizer/azure-ai-formrecognizer/samples/README.md:39]:
```suggestion
|[sample_manage_models.py][sample_manage_models] and [sample_manage_models_async.py][sample_manage_models_async]|Manage the models in your account|
```

**Review by maririos — APPROVED** [review]

File: sdk/formrecognizer/azure-ai-formrecognizer/samples/README.md [hunk 3]
```diff
@@ -33,6 +57,14 @@ All of these samples need the endpoint to your Form Recognizer resource ([instru
...
+|[sample_get_bounding_boxes.py][sample_get_bounding_boxes] and [sample_get_bounding_boxes_async.py][sample_get_bounding_boxes_async]|Get info to visualize the outlines of form content and fields, which can be used for manual validation|
+|[sample_differentiate_output_models_trained_with_and_without_labels.py][sample_differentiate_output_models_trained_with_and_without_labels] and [sample_differentiate_output_models_trained_with_and_without_labels_async.py][sample_differentiate_output_models_trained_with_and_without_labels_async]|See the differences in output when using a custom model trained with labeled data and one trained with unlabeled data|
+|[sample_differentiate_output_labeled_tables.py][sample_differentiate_output_labeled_tables] and [sample_differentiate_output_labeled_tables_async.py][sample_differentiate_output_labeled_tables_async]|See the differences in output when using a custom model trained with fixed vs. dynamic table tags|
+
+## Samples for client library versions 3.0.0 and below
+
+Please see the samples [here][v3.0.0-samples-tag].
 
 ## Prerequisites
 * Python 2.7, or 3.6 or later is required to use this package (3.6 or later if using asyncio)
```
Comment (catalinaperalta) [inline sdk/formrecognizer/azure-ai-formrecognizer/samples/README.md:70]: Do we need a disclaimer here as well or is the main README enough?

**Review by catalinaperalta — APPROVED** [review]
LGTM

File: sdk/formrecognizer/azure-ai-formrecognizer/samples/README.md (hunk 3 above)
Comment (kristapratico) [reply sdk/formrecognizer/azure-ai-formrecognizer/samples/README.md:70]: Do you mean the 2.7 disclaimer? I think laurent only said the main readme

**Pull request merged**